sunpy/coordinates/wcs_utils.py

```diff
@@ -74,6 +142,19 @@ def solar_wcs_frame_mapping(wcs):
                              attrs[2] * u.m,
                              **kwargs)
 
+    # Read the observer out of obsgeo for ground based observers
+    if observer is None:
+        try:
+            observer = obsgeo_to_frame(wcs.wcs.obsgeo, dateobs)
+            observer = SkyCoord(observer, rsun=rsun)
+        except ValueError as e:
```

Comment:
So if the value error happens, we assume this WCS is not ground based?

Reply:
yeah, this is also the last thing we check, so if this doesn't work we assume there isn't an observer defined.